azurerm/internal/services/iothub/iothub_resource.go

```diff
@@ -621,6 +635,13 @@ func resourceArmIotHubRead(d *schema.ResourceData, meta interface{}) error {
 		if err := d.Set("file_upload", fileUpload); err != nil {
 			return fmt.Errorf("Error setting `file_upload` in IoTHub %q: %+v", name, err)
 		}
+
+		if enabled := properties.PublicNetworkAccess; enabled != "" {
+			d.Set("public_network_access", false)
+			if enabled == devices.Enabled {
+				d.Set("public_network_access", true)
+			}
+		}
```

Comment:
couldn't we could make this:

```suggestion
		publicNetworkAccessEnabled := properties.PublicNetworkAccess == devices.Enabled
		d.Set("public_network_access", publicNetworkAccessEnabled)
```

Reply:
Sure could